### DISCUSSION TITLE
crossfade overlap time adjustable.

### DISCUSSION THREAD

**Discussion by compujunk** [music-assistant/support#1239] [Feature requests and ideas]
I would like to be able to adjust the crossfade overlap time again. I think the standard overlap that is now used is too much, but many differ, so it would be easier if everyone could make their own setting

**Comment by OzGav** [comment]
Noted .Will see what Marcel thinks in due course

**Comment by MangoCats** [comment]
I see this marked as implemented, but I am not finding how to adjust the overlap time in Server version 2.4.4 (latest official Docker image for amd64 as of April 1 2025...)

Is the control exposed yet?

**Reply by OzGav** [reply]
If the player supports it then it is in the advanced settings

**Comment by MangoCats** [comment]
Sorry, I am new to Music Assistant.  I feel like I have found at least a
dozen Advanced Settings menus so far, mostly just adjusting logging levels.

First, the output device is a stereo pair of Sonos Play:1 speakers, they
are added in Home Assistant because I wasn't getting them to play directly
without doing that.

Do you have any more detailed clues for where the setting would be found?
It's hard to know if I have looked in the right place if it's just missing
from an advanced setting, somewhere.

Thanks.

**Reply by OzGav** [reply]
You should be able to find these settings. First enable crossfade and then in the advanced section you will find the duration.

<img width="820" alt="image" src="https://github.com/user-attachments/assets/95418b58-cc40-472f-993f-62452bda5a09" />

Another person reported a problem with the Play 1 here https://github.com/music-assistant/support/issues/3829. but they aren’t using them in the stereo pair configuration.

**Reply by OzGav** [reply]
Please provide a screenshot which shows the same as what I show above.

**Comment by MangoCats** [comment]
Thanks!

So, I am guessing this means I don't have it?




Seems strange to me that Crossfade would work differently for different
output devices, I thought that any Crossfade used flow mode: pre-processing
the audio before sending it as a continuous stream?

**Comment by MangoCats** [comment]
As for the other users Sonos:Play 1 problem, I never had the "won't play
anything" issue.

What put me into Crossfade territory was that songs in the queue would be
skipped, same songs every time (so far) get skipped, but they all play
directly.  So, I found Crossfade and tried that, which I guess put the
output in "flow mode" and now the queue plays without bypassing certain
(like 35% of) songs.  But I don't see a way to select flow mode for
"gapless" playback either, which is how we are used to hearing our albums...

**Reply by OzGav** [reply]
So confirm you are actually using the Sonos MA provider. You said above 

> they are added in Home Assistant because I wasn't getting them to play directly without doing that.

**Comment by MangoCats** [comment]
So, I may have duplicated the Play 1 trouble ticket after all.  First try
with the Sonos option in MA didn't give any output, but HA was over there
all eager to add them so I let it and they started playing 99%
satisfactorily, only noticed the queue skipping issues after a while.

As far as I know I am using the Home Assistant device as output route for
Music Assistant.




Yes, and I did see that "strong recommendation" to use the player provider
that wasn't giving me any output at all:




If it would work, I would happily switch...

**Reply by OzGav** [reply]
Well that is the problem that should be resolved which is why I pointed out that other issue. 

Edit: what errors do you get in the MA log when you try and play to the device using the MA provider. Also have you updated the firmware on the device?

**Comment by MangoCats** [comment]
I agree, in principle, the Sonos direct driver should work.

In practice, Sonos:Play 1 via Home Assistant is 99% satisfactory for our
uses, and if the Crossfade time could be adjusted down to half or less of
its current behavior I would be 100% happy with the performance...

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
